### DISCUSSION TITLE
Mac M1: running uname -m returns `x86_64` when it previously returned `arm64` before rebooting.

### DISCUSSION THREAD

**Discussion by molleweide** [Homebrew/discussions#2226] [Everyday usage] [answered]
Hi,

I am running the Alacritty terminal on MacOS and I use brew to install.
After rebooting my M1 Macbook Air, running uname -m returns `x86_64`
when it previously returned `arm64` before rebooting.
If I check the same command in the regular Terminal.app. it returns
arm64 which is what it is supposed to do on M1 macs from what I understand.
Has anyone else stumbled upon this issue here? Is this an appropriate place
to ask for help on this issue?

I have build scripts on my M1 mac that depends on `uname -m` returning `arm64`
which worked perfectly until yesterday and now after having rebooted my scripts
don't work because they get `x86_64`.

Also I want to tell you that I am not an expert so I hope that I am not making
any stupid beginners mistake whilst reporting this. I will be digging further
to see what I could have done wrong. I am not sure if this is the right place
to ask for help.

**Comment by carlocab** [comment, marked as answer]
Alacritty is built for x86_64, so anything running inside of it will run under Rosetta. You'll need to wait for Alacritty to make a new release that is either a universal binary or built natively for M1.

**Reply by molleweide** [reply]
Thanks for replying. 

What I don't understand then is why it worked before rebooting? Do you have any idea how that
could work?

**Reply by carlocab** [reply]
No idea, unfortunately. I'm actually surprised `uname -m` ever showed `arm64` under Alacritty. Did you happen to have `coreutils` installed with `$HOMEBREW_PREFIX/opt/coreutils/libexec/gnubin` in your `PATH`? Because `coreutils`' `uname -m` will reflect the arch at build-time, and so will show `arm64` even when running in Rosetta.

**Reply by molleweide** [reply]
Thanks for writing this. I will check and get back later!

**Reply by molleweide** [reply]
Hi again carlocab!

If I run `which coreutils` on my M1 mac it returns not found.
Furthermore, I have homebrew_prefix set to `/opt/homebrew/`.

However, `/opt/Homebrew/opt/coreutils/libexec/gnubin` exists
on my system.

Should/can I add this to my path somehow without creating problems?

**Reply by molleweide** [reply]
Can I also ask you where I can read more about homebrew env variables?

**Reply by SMillerDev** [reply]
> If I run which coreutils on my M1 mac it returns not found.

coreutils isn't a command, it's a set of tools, `uname` is one of them.

> Should/can I add this to my path somehow without creating problems?

It will very likely create problems, due to the "feature" described by carlocab.

> Can I also ask you where I can read more about homebrew env variables?

The manpage: https://docs.brew.sh/Manpage#environment

**Reply by molleweide** [reply]
Hi thanks for replying and providing the link.

May I ask you if there is another way that I can check for `arm64` on my mac then
so that I can run my scripts without problems?

**Reply by SMillerDev** [reply]
It depends on what you want to know.

If you want to know if your script is executing on an ARM processor or on an x86 processor/emulator you can just use the macOS `uname`, which will tell you what architecture you're calling `uname` with. 

If instead you want to know if you're on an arm system or not you could check if `/opt/Homebrew` exists, since Homebrew installs in `/usr/local` on x86 systems. Or you could call `$(brew --prefix)/opt/coreutils/libexec/gnubin/uname` because it'll tell you on what architecture the software was compiled.

**Reply by molleweide** [reply]
i found a solution!

to use uname -a instead
